code/modules/mob/living/carbon/xenomorph/abilities/praetorian/praetorian_powers.dm

```diff
@@ -753,280 +753,174 @@
+	raging_valkyrie.armor_modifier += armor_buff
+	raging_valkyrie.recalculate_armor()
 
-		var/bonus_shield = 0
 
-		var/datum/behavior_delegate/praetorian_warden/behavior = wardenbuff_user.behavior_delegate
-		if (!istype(behavior))
-			return
 
-		if (!behavior.use_internal_hp_ability(shield_cost))
-			return
-
-		bonus_shield = behavior.internal_hitpoints*0.5
-		if (!behavior.use_internal_hp_ability(bonus_shield))
-			bonus_shield = 0
-
-		var/total_shield_amount = shield_amount + bonus_shield
-
-		if (wardenbuff_user.observed_xeno != null)
-			to_chat(wardenbuff_user, SPAN_XENOHIGHDANGER("We cannot shield [targetXeno] as effectively over distance!"))
-			total_shield_amount = total_shield_amount/4
-			targetXeno.visible_message(SPAN_BOLDNOTICE("[targetXeno]'s exoskeleton shimmers for a fraction of a second."))//marines probably should know if a xeno gets healed
-		else //so both visible messages don't appear at the same time
-			targetXeno.visible_message(SPAN_BOLDNOTICE("[wardenbuff_user] points at [targetXeno], and it shudders as its exoskeleton shimmers for a second!")) //this one is a bit less important than healing and rejuvenating
-		to_chat(wardenbuff_user, SPAN_XENODANGER("We bolster the defenses of [targetXeno]!")) //but i imagine it'll be useful for predators, survivors and for battle flavor
-		to_chat(targetXeno, SPAN_XENOHIGHDANGER("We feel our defenses bolstered by [wardenbuff_user]!"))
-
-		targetXeno.add_xeno_shield(total_shield_amount, XENO_SHIELD_SOURCE_WARDEN_PRAE, duration = shield_duration, decay_amount_per_second = shield_decay)
-		targetXeno.xeno_jitter(1 SECONDS)
-		targetXeno.flick_heal_overlay(3 SECONDS, "#FFA800") //D9F500
-		wardenbuff_user.add_xeno_shield(total_shield_amount*0.5, XENO_SHIELD_SOURCE_WARDEN_PRAE, duration = shield_duration, decay_amount_per_second = shield_decay) // X is the prae itself
-		wardenbuff_user.xeno_jitter(1 SECONDS)
-		wardenbuff_user.flick_heal_overlay(3 SECONDS, "#FFA800") //D9F500
-		use_plasma = TRUE
-
-	else if (curr_effect_type == WARDEN_HEAL_HP)
-		if (!wardenbuff_user.Adjacent(A))
-			to_chat(wardenbuff_user, SPAN_XENODANGER("We must be within touching distance of [targetXeno]!"))
-			return
-		if (SEND_SIGNAL(targetXeno, COMSIG_XENO_PRE_HEAL) & COMPONENT_CANCEL_XENO_HEAL)
-			to_chat(wardenbuff_user, SPAN_XENOWARNING("We cannot heal this xeno!"))
-			return
 
-		var/bonus_heal = 0
-		var/datum/behavior_delegate/praetorian_warden/behavior = wardenbuff_user.behavior_delegate
-		if (!istype(behavior))
-			return
 
-		if (!behavior.use_internal_hp_ability(heal_cost))
-			return
+	if(isxeno(buffing_target))
+		playsound(get_turf(buffing_target), "alien_roar", 40)
+		buffing_target.create_custom_empower(icolor = "#a31010", ialpha = 200, small_xeno = TRUE)
+		buffing_target.add_filter("raging", 1, list("type" = "outline", "color" = "#a31010", "size" = 1))
+		buffing_target.armor_modifier += target_armor_buff
+		buffing_target.recalculate_armor()
+		buffing_target = WEAKREF(focus_rage)
```

Comment:
What are you trying to do with the line `buffing_target = WEAKREF(focus_rage)` ? This only alters the local var, you're making a weakref to a weakref, and you no longer use it after this point. There is currently no point to make a weakref to the weakref focus_rage and assign that to the local var `buffing_target`
```suggestion
```